### DISCUSSION TITLE
How to call lncli create and unlock from api

### DISCUSSION THREAD

**Discussion by mentioneddd** [lightninglabs/lightning-terminal#738] [Troubleshooting]
Hi following problem, I'm currently developing a lightning wallet, in which every user should get assigned his/her individual node. However I currently struggle to setup the authentication process when users first join the app. I can spin up a docker running a node for each usee individually however how would I do the lncli create as an api call for the user if I dont want to put it in the dockerfile? Also is the docker and api setup mad  in a way where I dont need to mess with the dockerfile at all in order to spin up a docker instane for each user. Could I simply use only api calls in order to change stuff individually for each user?

My docker spinup currently looks something like this:

sudo docker run -d \
  -v ~/Desktop/wallet_password:/root/.lit/wallet_password \
  -v ~/Desktop/lit.conf:/root/.lit/lit.conf \
...
  -p 8443:8443/tcp \
  -p 9735:9735/tcp \
  -p 10009:10009/tcp \
  lightninglabs/lightning-terminal:v0.12.2-alpha --enablerest

Also can I instanly pass a seedphrase while spinning it up or does it need to be done with an api call or lncli create call in the dockerfile?
In general is there anything else you recommend or I should look at anything else trying to create a individual dockernode for each user?

**Comment by bitromortac** [comment]
https://lightning.engineering/api-docs/category/walletunlocker-service might be useful to you. You could also look into https://github.com/lightninglabs/lndinit
